control-plane/connect-inject/endpoints_controller_ent_test.go

```diff
@@ -299,12 +332,17 @@ func TestReconcileCreateEndpointWithNamespaces(t *testing.T) {
 			cfg.Namespace = testCase.ExpConsulNS
 			consulClient, err = api.NewClient(cfg)
 			require.NoError(t, err)
+
 			// After reconciliation, Consul should have the service with the correct number of instances.
 			serviceInstances, _, err := consulClient.Catalog().Service(setup.consulSvcName, "", &api.QueryOptions{Namespace: testCase.ExpConsulNS})
 			require.NoError(t, err)
-			service, _, err := consulClient.Catalog().Service("mesh-gateway", "", &api.QueryOptions{Namespace: "default"})
+			meshGatewayService, _, err := consulClient.Catalog().Service("mesh-gateway", "", &api.QueryOptions{Namespace: "default"})
+			require.NoError(t, err)
+			serviceInstances = append(serviceInstances, meshGatewayService...)
+			terminatingGatewayService, _, err := consulClient.Catalog().Service("terminating-gateway", "", &api.QueryOptions{Namespace: testCase.ExpConsulNS})
 			require.NoError(t, err)
-			serviceInstances = append(serviceInstances, service...)
+			serviceInstances = append(serviceInstances, terminatingGatewayService...)
+
```

Comment:
I wonder if there is a way this could be a touch more elegant. I don't like that each gateway has to be explicit :\

Reply:
The rebase mayhem unmoored this comment, but I did find a more scalable and elegant way to do this.